### DISCUSSION TITLE
Are CICD pipelines for applications supported in Gruntwork Pipelines?

### DISCUSSION THREAD

**Discussion by marijakstrazdas** [gruntwork-io/knowledge-base#103] [Help] [answered]
Are CICD pipelines for applications supported?

**Comment by yorinasub17** [comment, marked as answer]
Gruntwork Pipelines only supports the infrastructure component of application CI/CD. At its core, Gruntwork Pipelines is a framework for securely deploying infrastructure to AWS without having to grant access to CI servers direct wide ranging permissions to your AWS account.

To elaborate further, Gruntwork Pipelines provides scripts and solutions that allow you to take a built artifact from traditional CI pipelines, and deploy it using Terragrunt or Terraform. The bulk of the pipeline is driven by CI servers (e.g. CircleCI, GitLab, etc), which allows you to use many of the off the shelf pipeline code that is available in the community to cater to many CI use cases for your application.

For example, you could extend a standard CircleCI pipeline that implements the following workflow with Gruntwork Pipelines:

1. Run precommit checks
2. Run unit tests of application
3. [GRUNTWORK PIPELINES] Build docker image, tag with commit SHA, and push to ECR using AWS credentials in GW Pipelines, not in CircleCI.
4. [GRUNTWORK PIPELINES] Checkout infrastructure code and update the docker image tag for an ECS service in the `terragrunt.hcl`
5. [GRUNTWORK PIPELINES] Commit the updated `terragrunt.hcl`, push to `main`, and run `apply`.

Note that Gruntwork Pipelines does not contain off the shelf workflows for you to use, as many workflows are highly dependent and tightly coupled with how you organize your infrastructure code.

However, the Reference Architecture includes an off the shelf workflow that is compatible with the Reference Architecture, including template workflow configurations for the chosen CI server that can be installed on any application repo to be used to setup the above reference pipeline.
